lib/bloc/buy_bitcoin/moonpay/moonpay_bloc.dart

```diff
@@ -0,0 +1,84 @@
+import 'package:breez_sdk/breez_bridge.dart';
+import 'package:breez_sdk/bridge_generated.dart';
+import 'package:breez_translations/breez_translations_locales.dart';
+import 'package:c_breez/bloc/buy_bitcoin/moonpay/moonpay_state.dart';
+import 'package:c_breez/utils/exceptions.dart';
+import 'package:c_breez/utils/preferences.dart';
+import 'package:fimber/fimber.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:flutter_rust_bridge/flutter_rust_bridge.dart';
+
+final _log = FimberLog("MoonPayBloc");
+
+class MoonPayBloc extends Cubit<MoonPayState> {
+  final BreezBridge _breezLib;
+  final Preferences _preferences;
+
+  MoonPayBloc(
+    this._breezLib,
+    this._preferences,
+  ) : super(MoonPayState.initial());
+
+  Future<void> fetchMoonpayUrl() async {
+    _log.v("fetchMoonpayUrl");
+    emit(MoonPayState.loading());
+
+    try {
+      final url = await _breezLib.buyBitcoin(BuyBitcoinProvider.MoonPay);
+      _log.v("fetchMoonpayUrl url: $url");
+      emit(MoonPayState.urlReady(url));
+    } catch (e) {
+      _log.e("fetchMoonpayUrl error: $e");
+
+      if (e is FfiException && e.message.startsWith("Swap in progress was detected for address")) {
```

Comment:
I don't think we should check for ffiexception or specific string comes from the sdk, this tends to change and it will break the app.
I would:
1. check for inprogress first and show it if exists.
2. then try optimisticly to fetch new url and just show the error using the extractExceptionMessage